app.py

```diff
@@ -36,8 +43,90 @@ def datasets(index: str):
 @app.route('/datasets/refresh')
 def refresh_datasets():
     global masader
+    global embeddings
+    global cluster
 
     print('Refreshing the dataset...')
-    masader = load_masader_dataset_as_dict()
+    data = list(load_masader_dataset_as_dict())
+    masader = data
+    embeddings = get_embeddings_data(data)
+    cluster = get_cluster_data()
 
     return jsonify(f'The datasets updated successfully! The current number of available datasets is {len(masader)}.')
+
+
+# -----------Clusters-------------
+
+@app.route('/cluster')
+def cluster_req():
+    return jsonify(cluster)
+
+
+def get_cluster_data():
+    global tsne_data
+
+    clustering = KMeans(n_clusters=15).fit(tsne_data)
+    return pd.DataFrame(clustering.labels_.reshape(len(tsne_data), 1)).to_json()
+
+
+# -----------Embeddings-------------
+
+@app.route('/embeddings')
```

Comment:
Do we really need this endpoint? I think we need the `/clusters` endpoint only.

Reply:
Not really, just if, for some reason, it was needed to only retrieve the clusters.

Reply:
I think it is better to remove it if we don't need it, as you know, we have one file for the endpoints.